data-prepper-plugins/anomaly-detector-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/anomalydetector/AnomalyDetectorProcessor.java

```diff
@@ -30,14 +33,19 @@ public class AnomalyDetectorProcessor extends AbstractProcessor<Record<Event>, R
     public static final String DEVIATION_KEY = "deviation_from_expected";
     public static final String GRADE_KEY = "grade";
     static final String NUMBER_RCF_INSTANCES = "RCFInstances";
+    static final String CARDINALITY_OVERFLOW = "cardinalityOverflow";
 
     private final Boolean verbose;
+    private final int cardinalityLimit;
     private final IdentificationKeysHasher identificationKeysHasher;
     private final List<String> keys;
     private final PluginFactory pluginFactory;
     private final HashMap<Integer, AnomalyDetectorMode> forestMap;
     private final AtomicInteger cardinality;
     private final AnomalyDetectorProcessorConfig anomalyDetectorProcessorConfig;
+    private static final Logger LOG = LoggerFactory.getLogger(AnomalyDetectorProcessor.class);
+    private final Counter cardinalityOverflowCounter;
+    private boolean overflowWarned = false;
```

Comment:
It may not be a bad idea to make this some sort of timer or "last logged timestamp." As this is never set back to false, this will log once. So if this trips again a week later, you don't get the log.

```
Instant nextWarnTime = Instant.MIN;

...

        if(Instant.now().isAfter(nextWarnTime)) {
            ...
            nextWarnTime = Instant.now().plus(5, ChronoUnit.MINUTES);
        }
```